backend/src/test/java/de/bund/digitalservice/ris/norms/adapter/input/restapi/controller/ElementControllerTest.java

```diff
@@ -32,45 +34,273 @@ class ElementControllerTest {
-
-    // when
-    mockMvc
-        .perform(get(url))
+      @Test
+      void returns404IfNormNotFoundByEli() throws Exception {
+        // given no norm
+        // when
+        mockMvc
+            .perform(
+                get("/api/v1/norms/eli/bund/NONEXISTENT_NORM/1964/s593/1964-08-05/1/deu/regelungstext-1/elements/hauptteil-1_art-3")
+                    .accept(MediaType.TEXT_HTML))
+            .andExpect(status().isNotFound());
         // then
-        .andExpect(status().isOk())
-        // preface
-        .andExpect(jsonPath("$[0]").exists())
-        .andExpect(jsonPath("$[0].title").value("Dokumentenkopf"))
-        .andExpect(jsonPath("$[0].eid").value("einleitung-1"))
-        .andExpect(jsonPath("$[0].type").value("preface"))
-        // preamble
-        .andExpect(jsonPath("$[1].title").value("Eingangsformel"))
-        .andExpect(jsonPath("$[1].eid").value("preambel-1"))
-        .andExpect(jsonPath("$[1].type").value("preamble"))
-        // articles
-        .andExpect(jsonPath("$[2].title").value("Artikel 1 Änderung des Vereinsgesetzes"))
-        .andExpect(jsonPath("$[2].eid").value("hauptteil-1_art-1"))
-        .andExpect(jsonPath("$[2].type").value("article"))
-        .andExpect(jsonPath("$[3].title").value("Artikel 3 Inkrafttreten"))
-        .andExpect(jsonPath("$[3].eid").value("hauptteil-1_art-3"))
-        .andExpect(jsonPath("$[3].type").value("article"))
-        // conclusion
-        .andExpect(jsonPath("$[4].title").value("Schlussteil"))
-        .andExpect(jsonPath("$[4].eid").value("schluss-1"))
-        .andExpect(jsonPath("$[4].type").value("conclusions"));
+      }
+
+      @Test
+      void returns404IfElementNotFoundByEid() throws Exception {
+        // given
+        var norm = NormFixtures.loadFromDisk("NormWithMultipleMods.xml");
+        when(loadNormUseCase.loadNorm(any())).thenReturn(Optional.of(norm));
+
+        // when
+        mockMvc
+            .perform(
+                get("/api/v1/norms/eli/bund/bgbl-1/2017/s419/2017-03-15/1/deu/regelungstext-1/elements/NONEXISTENT_EID")
+                    .accept(MediaType.TEXT_HTML))
+            // then
+            .andExpect(status().isNotFound());
+      }
+
+      @Test
+      void returnsBadRequestIfAtIsoDateIsInvalid() throws Exception {
+        // Given
+
+        // When / Then
+        mockMvc
+            .perform(
+                get("/api/v1/norms/eli/bund/bgbl-1/1990/s2954/2022-12-19/1/deu/regelungstext-1/elements/hauptteil-1_para-20?atIsoDate=INVALID")
+                    .accept(MediaType.TEXT_HTML))
+            .andExpect(status().isBadRequest());
+      }
+    }
+
+    @Test
+    void returnsHtmlRendering() throws Exception {
+      // given
+      var norm = NormFixtures.loadFromDisk("NormWithMultipleMods.xml");
+      when(loadNormUseCase.loadNorm(any())).thenReturn(Optional.of(norm));
+      var renderedHtml = "renderedHtml";
+      when(transformLegalDocMlToHtmlUseCase.transformLegalDocMlToHtml(any()))
```

Comment:
Idea: instead of using `any` for the query, use a more specific query so the mock only returns data if it is called correctly.

Reply:
Done